lib/mix/tasks/openapi.spec.json.ex

```diff
@@ -21,14 +21,18 @@ defmodule Mix.Tasks.Openapi.Spec.Json do
     write_spec(content, opts.filename)
   end
 
-  def generate_spec(%{spec: spec, pretty: pretty}) do
+  def generate_spec(%{spec: spec, pretty: pretty, vendor_extensions: vendor_extensions}) do
     case Code.ensure_compiled(spec) do
       {:module, _} ->
         if function_exported?(spec, :spec, 0) do
           json_encoder = OpenApiSpex.OpenApi.json_encoder()
           spec = spec.spec()
 
-          case json_encoder.encode(spec, pretty: pretty) do
+          json_encoding_result = spec
+          |> OpenApiSpex.OpenApi.to_map(vendor_extensions: vendor_extensions)
+          |> OpenApiSpex.OpenApi.json_encoder().encode(pretty: pretty)
```

Comment:
To fix the unused variable warning: 

```suggestion
          |> json_encoder.encode(pretty: pretty)
```